core-bundle/src/Resources/contao/library/Contao/Template.php

```diff
@@ -62,6 +62,8 @@
  * @property boolean      $trustedDevicesEnabled
  * @property array        $trustedDevices
  * @property string       $currentDevice
+ *
+ * @deprecated this class will be removed in Contao 6, use Twig templates instead
```

Comment:
Do we really want to deprecate the `Template` class? It will generate a lot of deprecation warnings in IDEs such as PHPStorm and people cannot really do anything about it as long as .html5 templates are in use.

Reply:
>  as long as .html5 templates are in use.

But the plan was to deprecate them in 5 anyway, no?

Reply:
Well, currently it's only a soft deprecation, not one that would trigger an error. IMHO there are no cases where you need to instantiate a template yourself although the core does. But we could also postpone this discussion and see what's the state close to the release? Wdyt?

Reply:
Agreed.